### PR TITLE
feat(strm-1259): remove billing id usage

### PR DESCRIPTION
**Pull request by bobvandenhoogen**
BREAKING CHANGE: authentication has changed

### CODE REVIEW

File: lib/src/main/java/io/strmprivacy/driver/client/AuthService.java [hunk 1]
```diff
@@ -86,37 +83,39 @@ public void stop() {
...
-            authenticate(billingId, clientId, clientSecret);
+            authenticate(clientId, clientSecret);
         }
     }
 
     private void doPost(String uri, ObjectNode payload) throws IOException, InterruptedException, TimeoutException, ExecutionException {
         ContentResponse response = httpClient.POST(uri)
                 .content(new StringContentProvider(MAPPER.writeValueAsString(payload)))
-                .header(HttpHeader.CONTENT_TYPE, "application/json; charset=UTF-8")
+                .header("content-type", "application/x-www-form-urlencoded")
```
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthService.java]: You can use the `HttpHeader` constants.

File: lib/src/main/java/io/strmprivacy/driver/domain/Config.java [hunk 2]
```diff
@@ -6,21 +6,14 @@ public class Config {
...
     private Config(Builder builder) {
         setGatewayScheme(builder.gatewayScheme);
         setGatewayHost(builder.gatewayHost);
         setGatewayEndpoint(builder.gatewayEndpoint);
         setGatewayPort(builder.gatewayPort);
-        setStsScheme(builder.stsScheme);
-        setStsHost(builder.stsHost);
-        setStsAuthEndpoint(builder.stsAuthEndpoint);
-        setStsRefreshEndpoint(builder.stsRefreshEndpoint);
+        setKeycloakUrl(builder.keycloakUrl);
```
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/domain/Config.java]: For consistency sake, please split the URL parts, as before.

File: lib/src/test/java/io/strmprivacy/driver/client/StrmPrivacyClientTest.java [hunk 3]
```diff
@@ -61,6 +59,7 @@ private Config getTestConfig(WireMockRuntimeInfo wireMockRuntimeInfo) {
                 .gatewayHost("localhost")
                 .gatewayPort(wireMockRuntimeInfo.getHttpPort())
                 .gatewayScheme("http")
+                .keycloakUrl("https://accounts.dev.strmprivacy.io/auth/realms/streams/protocol/openid-connect/token")
```
Comment (trietsch) [inline lib/src/test/java/io/strmprivacy/driver/client/StrmPrivacyClientTest.java]: Did you actually update the test itself? You should create the specific mocks with Wiremock in order to properly test it.

**Review by trietsch — CHANGES_REQUESTED** [review]

File: lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java [hunk 4]
```diff
@@ -1,7 +1,7 @@
 package io.strmprivacy.driver.client;
 
 class AuthProvider {
-    private String idToken;
+    private String access_token;
```
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java]: Use pascal case.

File: lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java [hunk 5]
```diff
@@ -10,18 +10,18 @@ public AuthProvider() {
...
-        this.idToken = idToken;
+    public AuthProvider(String access_token, String refreshToken, long expiresAt) {
+        this.access_token = access_token;
         this.refreshToken = refreshToken;
         this.expiresAt = expiresAt;
     }
 
-    public String getIdToken() {
-        return idToken;
+    public String getAccess_token() {
```
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthProvider.java]: casing?

File: lib/src/main/java/io/strmprivacy/driver/domain/Config.java [hunk 6]
```diff
@@ -58,20 +62,39 @@ public void setGatewayPort(int gatewayPort) {
...
+
     public static final class Builder {
         private String gatewayScheme = "https";
         private String gatewayHost = "events.strmprivacy.io";
         private String gatewayEndpoint = "/event";
         private int gatewayPort = 443;
-        private String keycloakUrl = "https://accounts.strmprivacy.io/auth/realms/streams/protocol/openid-connect/token";
+        private String keycloakScheme = "https";
+        private String keycloakHost = "accounts.strmprivacy.io";
+        private String keycloakEndpoint = "auth/realms/streams/protocol/openid-connect/token";
```
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/domain/Config.java]: Convention is to start paths on a host with `/`.

File: lib/src/main/java/io/strmprivacy/driver/client/AuthService.java [hunk 7]
```diff
@@ -53,8 +53,8 @@ public AuthService(String clientId, String clientSecret, Config config) {
         }
 
         try {
-            this.authUri = new URI(config.getKeycloakUrl()).toString();
-            this.refreshUri = new URI(config.getKeycloakUrl()).toString();
+            this.authUri = new URI(String.format("%s://%s/%s", config.getKeycloakScheme(), config.getKeycloakHost(), config.getKeycloakEndpoint())).toString();
```
Comment (trietsch) [inline lib/src/main/java/io/strmprivacy/driver/client/AuthService.java]: The slash should be removed here for the path. That should be included in the path parameter.

**Review by trietsch — CHANGES_REQUESTED** [review]

**Review by trietsch — CHANGES_REQUESTED** [review]
~Once we can merge this, **this PR should be merged as breaking change**~ -> no changes required.

**Review by trietsch — APPROVED** [review]
Approved, but once we can merge this, **this PR should be merged as breaking change**

**Review by ivan-p92 — APPROVED** [review]

**Pull request merged**